CS3230Project/DAL/Nurses/NurseDal.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using CS3230Project.Model.Users;
+using MySql.Data.MySqlClient;
+
+namespace CS3230Project.DAL.Nurses
+{
+    /// <summary>
+    ///   The data access layer for the nurses
+    /// </summary>
+    public static class NurseDal
+    {
+
+        /// <summary>
+        /// Gets the nurses.
+        ///
+        /// Precondition: none
+        /// Post-condition: none
+        /// </summary>
+        /// <returns>
+        ///   A list of all the nurses
+        /// </returns>
+        public static List<Nurse> GetNurses()
+        {
+            using var connection = new MySqlConnection(Connection.ConnectionString);
+            connection.Open();
+            const string query = "select * from nurses";
+            using var command = new MySqlCommand(query, connection);
+
+            return NurseDal.createNurses(command);
+        }
+
+        /// <summary>
+        /// Gets the nurse by identifier.
+        ///
+        /// Precondition: none
+        /// Post-condition: none
+        /// </summary>
+        /// <param name="ID">The identifier.</param>
+        /// <returns>
+        ///   the nurse with the provided ID
+        /// </returns>
+        /// <exception cref="System.ArgumentException"></exception>
+        public static Nurse GetNurseByID(int ID)
+        {
+            Nurse nurse = null;
+            using var connection = new MySqlConnection(Connection.ConnectionString);
+            connection.Open();
+            const string query = "select * from nurses where nurseId = @nurseID";
+            using var command = new MySqlCommand(query, connection);
+            command.Parameters.Add("@nurseId", MySqlDbType.String).Value = ID;
+
+            using var reader = command.ExecuteReader();
+            var nurseIDOrdinal = reader.GetOrdinal("nurseId");
+            var employeeIDOrdinal = reader.GetOrdinal("employeeId");
+            var firstNameOrdinal = reader.GetOrdinal("firstName");
+            var lastNameOrdinal = reader.GetOrdinal("lastName");
+            var dateOfBirthOrdinal = reader.GetOrdinal("dateOfBirth");
+            var genderOrdinal = reader.GetOrdinal("gender");
+            var phoneOrdinal = reader.GetOrdinal("phone");
+            var addressOneOrdinal = reader.GetOrdinal("addressOne");
+            var addressTwoOrdinal = reader.GetOrdinal("addressTwo");
+            var cityOrdinal = reader.GetOrdinal("city");
+            var stateOrdinal = reader.GetOrdinal("state");
+            var zipcodeOrdinal = reader.GetOrdinal("zipcode");
+            var usernameOrdinal = 2; //will be removed after merge, I think.
```

Comment:
What is this, looks like this  is a temporary thing that should be removed.

Reply:
Looking at this more we cannot just set the userName to 2

Reply:
This is a temp thing. It will be removed when I finish the Nurse class. I had planned to do that after the merge.

Reply:
I can make those changes now that the branches are merged.

Reply:
Fixed